asserts/snap_asserts.go

```diff
@@ -698,3 +698,121 @@ func InitBuiltinBaseDeclaration(headers []byte) error {
+	Until     time.Time
+}
+
+// SnapDeveloper holds a snap-developer assertion, defining the developers who
+// can collaborate on a snap while it's owned by a specific publisher.
+type SnapDeveloper struct {
+	assertionBase
+	developers []*Developer
+}
+
+// SnapID returns the snap id of the snap.
+func (snapdev *SnapDeveloper) SnapID() string {
+	return snapdev.HeaderString("snap-id")
+}
+
+// PublisherID returns the publisher's account id.
+func (snapdev *SnapDeveloper) PublisherID() string {
+	return snapdev.HeaderString("publisher-id")
+}
+
+// Developers returns the developers allowed to collaborate on the snap.
+func (snapdev *SnapDeveloper) Developers() []*Developer {
+	return snapdev.developers
+}
+
+func (snapdev *SnapDeveloper) checkConsistency(db RODatabase, acck *AccountKey) error {
+	// Check authority is the publisher or trusted.
+	authorityID := snapdev.AuthorityID()
+	publisherID := snapdev.PublisherID()
+	if !db.IsTrustedAccount(authorityID) && (publisherID != authorityID) {
+		return fmt.Errorf("snap-developer must be signed by the publisher or a trusted authority: authority-id=%q, publisher-id=%q", authorityID, publisherID)
+	}
+
+	_, err := db.Find(SnapDeclarationType, map[string]string{
+		// XXX: mediate getting current series through some context object? this gets the job done for now
+		"series":  release.Series,
+		"snap-id": snapdev.SnapID(),
+	})
+	if err == ErrNotFound {
+		return fmt.Errorf("snap-developer assertion for snap id %q does not have a matching snap-declaration assertion", snapdev.SnapID())
+	}
+
+	// TODO(matt): check publisher-id exists (only needed if different to account-id")
+	// TODO(matt): check that snap-decl's publisher is snap-dev's publisher? not sure about this.
+	// TODO(matt): check each developer id has an acccount assertion.
+
+	return nil
+}
+
+// sanity
+var _ consistencyChecker = (*SnapDeveloper)(nil)
+
+// Prerequisites returns references to this snap-developer's prerequisite assertions.
+func (snapdev *SnapDeveloper) Prerequisites() []*Ref {
+	// TODO(matt):
+	// - account - developers and publisher
+	// - snap-declaration
+	return nil
+}
+
+func assembleSnapDeveloper(assert assertionBase) (Assertion, error) {
+	developers, err := checkDevelopers(assert.headers, "developers")
+	if err != nil {
+		return nil, err
+	}
+
+	return &SnapDeveloper{
+		assertionBase: assert,
+		developers:    developers,
+	}, nil
+}
+
+func checkDevelopers(headers map[string]interface{}, name string) ([]*Developer, error) {
+	// TODO:
+	// - reject overlapping date ranges?
+	value, ok := headers[name]
+	if !ok {
+		return nil, nil
+	}
+	list, ok := value.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("%q must be a list of developer maps", name)
+	}
+	if len(list) == 0 {
+		return nil, nil
+	}
+
+	developers := make([]*Developer, len(list))
+	for i, item := range list {
+		v, ok := item.(map[string]interface{})
+		if !ok {
+			return nil, fmt.Errorf("%q must be a list of developers", name)
```

Comment:
Good catch, I missed this when I changed the other message ... and it turns out that wasn't tested.